ui/src/app/pages/appAnalysis/apps-list.tsx

```diff
@@ -9,40 +10,137 @@ import { Button } from 'app/components/buttons';
 import { ListPageHeader } from 'app/components/headers';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
 import colors from 'app/styles/colors';
-import { withCurrentWorkspace } from 'app/utils';
+import { reactStyles, withCurrentWorkspace } from 'app/utils';
 import { AnalyticsTracker } from 'app/utils/analytics';
+import { APP_LIST, JUPYTER_APP } from 'app/utils/constants';
+import { WorkspacePermissionsUtil } from 'app/utils/workspace-permissions';
+
+const styles = reactStyles({
+  fadeBox: {
+    margin: 'auto',
+    marginTop: '1rem',
+    width: '95.7%',
+  },
+  startButton: {
+    paddingLeft: '0.5rem',
+    height: '2rem',
+    backgroundColor: colors.secondary,
+  },
+  appsLabel: {
+    color: colors.primary,
+    fontWeight: 600,
+    fontSize: '14px',
+    lineHeight: '24px',
+    paddingBottom: '0.5rem',
+  },
+});
 
 export const AppsList = withCurrentWorkspace()((props) => {
   const { workspace } = props;
+  const [selectedApp, setSelectedApp] = useState('');
+  const [showAppModal, setShowAppModal] = useState(false);
+  const [showJupyterModal, setShowJupyterModal] = useState(false);
+
+  const displayJupyterModal = () => {
+    setShowAppModal(false);
+    setShowJupyterModal(true);
+  };
+
+  const canWrite = (): boolean => {
+    return WorkspacePermissionsUtil.canWrite(props.workspace.accessLevel);
+  };
+
+  const showApplicationModal = () => {
+    switch (selectedApp) {
+      case JUPYTER_APP: {
+        displayJupyterModal();
+        break;
+      }
+    }
+  };
+
+  const onClose = () => {
+    setShowJupyterModal(false);
+    setShowAppModal(false);
+  };
 
   useEffect(() => {
     props.hideSpinner();
   }, []);
 
   return (
-    <FadeBox style={{ margin: 'auto', marginTop: '1rem', width: '95.7%' }}>
-      <FlexColumn>
-        <FlexRow>
-          <ListPageHeader style={{ paddingRight: '1.5rem' }}>
-            Your Analysis
-          </ListPageHeader>
-          <Button
-            style={{
-              paddingLeft: '0.5rem',
-              height: '2rem',
-              backgroundColor: colors.secondary,
-            }}
-            onClick={() => {
-              AnalyticsTracker.Notebooks.OpenCreateModal();
-            }}
-            disabled={workspace.billingStatus === BillingStatus.INACTIVE}
-          >
-            <div style={{ paddingRight: '0.5rem' }}>Start</div>
-            <FontAwesomeIcon icon={faPlusCircle}></FontAwesomeIcon>
-          </Button>
-        </FlexRow>
-      </FlexColumn>
-    </FadeBox>
+    <>
+      <FadeBox style={styles.fadeBox}>
+        <FlexColumn>
+          <FlexRow>
+            <ListPageHeader style={{ paddingRight: '1.5rem' }}>
+              Your Analysis
+            </ListPageHeader>
+            <Button
+              data-test-id='start-button'
+              style={styles.startButton}
+              onClick={() => {
+                AnalyticsTracker.Notebooks.OpenCreateModal();
```

Comment:
is "Open Create Notebook Modal" the correct action for analytics here?

Reply:
You are right this should be at the end of the flow part of story 8975